scripts/functions/requirements/osx_brew

```diff
@@ -331,7 +331,7 @@ requirements_osx_brew_define_openssl()
       ;;
 
     (*)
-      brew_openssl_package="openssl@1.1"
+      brew_openssl_package="openssl@3"
```

Comment:
This depends on the version of Ruby. Only latest can work with v3. Can you add a switch case detecting Ruby version and installing appropriate dependency?

Reply:
Ah, right. Yeah I can do that! I'm thinking the switch case should be updated so `openssl@3` is used for all Ruby versions (major and minor) going forward? There's probably a cleaner way to do this, but something like this for example (leaving the rest of the switch case as is)?

```
case "$1" in
  (ruby-3.[2-9]*|ruby-3.[1-9][0-9]*|ruby-[4-9]*)
    brew_openssl_package="openssl@3"
    ;;

  (ruby-2.3*|ruby-2.2*|ruby-2.1*|ruby-2.0*|ruby-1.9*|ruby-1.8*)
    brew_openssl_package="openssl"
    ;;

  (ree-1.8*)
    brew_openssl_package="openssl"
    ;;

  (rbx-1*|rbx-2*)
    brew_openssl_package="openssl"
    ;;

  (truffleruby*)
    brew_openssl_package="openssl"
    ;;

  (*)
    brew_openssl_package="openssl@1.1"
    ;;
esac
```

Or is there some way to clean up the switch case so `openssl@3` is the new default? This wouldn't be quite right, but just for example:

```
case "$1" in
    ruby-2.3*|ruby-2.2*|ruby-2.1*|ruby-2.0*|ruby-1.9*|ruby-1.8*)
        brew_openssl_package="openssl"
        ;;
    
    ree-1.8*)
        brew_openssl_package="openssl"
        ;;
    
    rbx-1*|rbx-2*)
        brew_openssl_package="openssl"
        ;;
    
    truffleruby*)
        brew_openssl_package="openssl"
        ;;
    
    ruby-3.0*|ruby-3.1*)
        brew_openssl_package="openssl@1.1"
        ;;
    
    *)
        brew_openssl_package="openssl@3"
        ;;
esac
```

Also, is using `openssl` for these other cases still correct? For example, I think this:

```
(ruby-2.3*|ruby-2.2*|ruby-2.1*|ruby-2.0*|ruby-1.9*|ruby-1.8*)
  brew_openssl_package="openssl"
  ;;
```

Would actually install `openssl@3` now? The new default has been updated [to be openssl@3](https://github.com/Homebrew/homebrew-core/blob/master/Aliases/openssl) I think? Sorry for the questions! Just working through some uncertainty.